network/p2p/gossip/gossip.go

```diff
@@ -34,20 +38,54 @@ func NewGossiper[T any, U GossipableAny[T]](
 	log logging.Logger,
 	set Set[U],
 	client *p2p.Client,
-) *Gossiper[T, U] {
-	return &Gossiper[T, U]{
+	metrics prometheus.Registerer,
+	namespace string,
+) (*Gossiper[T, U], error) {
+	receivedTime, err := metric.NewAverager(
+		namespace,
+		"gossip_received_time",
+		"gossip response handling latency (ns)",
+		metrics,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	g := &Gossiper[T, U]{
 		config: config,
 		log:    log,
 		set:    set,
 		client: client,
+		receivedN: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "gossip_received_n",
+			Help:      "amount of gossip received (n)",
+		}),
+		receivedBytes: prometheus.NewCounter(prometheus.CounterOpts{
+			Namespace: namespace,
+			Name:      "gossip_received_bytes",
+			Help:      "amount of gossip received (bytes)",
+		}),
+		receivedTime: receivedTime,
 	}
+
+	errs := wrappers.Errs{}
+	errs.Add(
+		metrics.Register(g.receivedN),
+		metrics.Register(g.receivedBytes),
```

Comment:
does `receivedTime` need to be registered?

Reply:
It's registered internally in https://github.com/ava-labs/avalanchego/blob/3d1040dc26bea0985c339f0876e6dd338ac8e234/utils/metric/averager.go#L46